### PR TITLE
Allow passing public key using env variable

### PR DESCRIPTION
**Pull request by nanaya**
Mainly for using it as docker service in github action (although it also means specifying the full key in yaml file). Also just simpler overall setup with no mounted volume.

On unrelated note, the token scope must be either `*` or `read` of which the latter doesn't exist anymore ( ﾟ ヮﾟ)a

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -68,6 +70,9 @@ const config: Config = {
   },
   debug: process.env.APP_DEBUG === 'true',
   env,
+  oauthPublicKey: process.env.PASSPORT_PUBLIC_KEY == null
+    ? fs.readFileSync(`${baseDir}/oauth-public.key`)
+    : Buffer.from(process.env.PASSPORT_PUBLIC_KEY),
```
Comment (ThePooN) [inline src/config.ts:75]: this file contains newlines, can be tricky to deal with in env vars in some container/orchestration systems. i'd use base64 encoding for broader compatbility, although the file is already base64 encoded in the first place

**Review by ThePooN — CHANGES_REQUESTED** [review]

File: src/config.ts (hunk 1 above)
Comment (nanaya) [reply src/config.ts:75]: this is the default for laravel and if the tool have problem handing it then there may be problem with the tool as it's just standard env file. A bunch of tools/libraries also have been updated to handle newlines specifically to deal with embedding keys.

Comment (nanaya) [reply src/config.ts:75]: also fwiw, replacing the newlines with `\n` should also work

Comment (ThePooN) [reply src/config.ts:75]: my opinion isn't too strong about it, and the Helm chart already handles it by mounting the var to a file anyway. just a consideration for you, as newlines in envars aren't common place and they make masking secrets in logs more difficult, although i don't believe that will be an issue here and it's the public key anyway.

Comment (nanaya) [reply src/config.ts:75]: there shouldn't be problem from dev side. And masking secrets in log line by line sounds like looking for trouble in the first place.

Comment (ThePooN) [reply src/config.ts:75]: filtering out secrets from logs output is very common in enterprise

**Review by ThePooN — APPROVED** [review]

**Pull request merged**